.github/workflows/test.yml

```
@@ -0,0 +1,125 @@
+
+on:
+  push:
+    branches: main
+  pull_request:
+    branches: main
+  workflow_dispatch:
+  workflow_call:
+
+jobs:
+  setup-conda-build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+    
+      - name: Set up micromamba
+        uses: mamba-org/setup-micromamba@v2
+        with:
+          environment-file: ci/environment.yml
+          cache-environment-key: environment-${{ hashFiles('ci/environment.yml') }}
+
+  test-action:
+    name: Test ${{ matrix.id }}
+    runs-on: ubuntu-latest
+    needs: setup-conda-build
+    concurrency:
+      group: branch-${{ github.ref }}-${{ matrix.id }} # Cancel past jobs coming from the same branch
+      cancel-in-progress: true
+    env:
+      COMMON_INPUTS: >-
+        "meta_yaml_dir": ci/asyncio_recipe,
+        "upload": false
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - id: 1
+            label: "Test: Conversion to all platforms, 'conda_build_args' argument, '.conda' package format."
+            inputs: >-
+                "platform_all": true,
+                "conda_build_args": --package-format .conda
+            expected_outputs: >-
+              linux-32/asyncio-3.4.1-py311_0.conda
+              linux-64/asyncio-3.4.1-py311_0.conda
+              linux-aarch64/asyncio-3.4.1-py311_0.conda
+              linux-armv6l/asyncio-3.4.1-py311_0.conda
+              linux-armv7l/asyncio-3.4.1-py311_0.conda
+              linux-ppc64/asyncio-3.4.1-py311_0.conda
+              linux-ppc64le/asyncio-3.4.1-py311_0.conda
+              linux-s390x/asyncio-3.4.1-py311_0.conda
+              osx-64/asyncio-3.4.1-py311_0.conda
+              osx-arm64/asyncio-3.4.1-py311_0.conda
+              win-32/asyncio-3.4.1-py311_0.conda
+              win-64/asyncio-3.4.1-py311_0.conda
+              win-arm64/asyncio-3.4.1-py311_0.conda
+          - id: 2
+            label: "Test: Conversion to specific platforms, 'conda_convert_args' argument, '.tar.bz2' package format."
+            inputs: >-
+                "platform_win-32": true,
+                "platform_win-64": true,
+                "platform_osx-64": true,
+                "conda_convert_args": --platform win-32 -p osx-arm64,
+                "conda_build_args": --package-format .tar.bz2
+            expected_outputs: >-
+              linux-64/asyncio-3.4.1-py311_0.tar.bz2
+              osx-64/asyncio-3.4.1-py311_0.tar.bz2
+              osx-arm64/asyncio-3.4.1-py311_0.tar.bz2
+              win-32/asyncio-3.4.1-py311_0.tar.bz2
+              win-64/asyncio-3.4.1-py311_0.tar.bz2
+          - id: 3
+            label: "Test: 'platform_host: false', 'anaconda_upload_args' argument."
+            inputs: >-
+                "platform_host": false,
+                "anaconda_upload_args": --label mylabel
+            expected_outputs: ""
+          - id: 4
+            label: "Test: 'platform_host: false', 'anaconda_upload_args' argument."
```

```suggestion
          - id: 3
            label: "Test: 'keep_host_platform: false', 'anaconda_upload_args' argument."
            inputs: >-
                "keep_host_platform": false,
                "anaconda_upload_args": --label mylabel
            expected_outputs: ""
          - id: 4
            label: "Test: 'conda_convert_args' platform argument, 'anaconda_upload_args' argument."
```

The tests just need a small update with `keep_platform_host`

---

Ooops thank you!
Addressed in [518736d40604c4369c214022157efcebc68892dd](https://github.com/ACCESS-NRI/action-build-and-upload-conda-packages/pull/1/commits/518736d40604c4369c214022157efcebc68892dd)
I also revised the test labels.